File: C++/clinicaVeterinaria/Animal.cpp
#include <iostream>
#include "Animal.h"

using namespace std;
/*************** CONSTRUCTORES ***************/
/*Constructor por default de la clase Animal*/
Animal::Animal(){
    nombre = "undefined";
    enfermo = 0;
}
/*Constructor con argumentos para la clase Animal*/
Animal::Animal(string n, Fecha f ,int e){
    nombre = n;
    fechaNacimiento = f;
    enfermo = e;
}

/************* GETTERS Y SETTERS *************/

string Animal::getNombre(){
    return nombre;
}

Fecha Animal::getFechaNacimiento(){
    return fechaNacimiento;
}

int Animal::getEnfermo(){
    return enfermo;
}

int Animal::setNombre( string n ){
    nombre = n;
    return 1;
}

int Animal::setFechaNacimiento( Fecha f ){
    fechaNacimiento = f;
    return 1;
}

/*1-enfermo y 0-sano*/
int Animal::setEnfermo(int e){
    enfermo = e;
    return 1;
}

/************ METODOS DE FUNCIONALIDAD ***********/

void Animal::sanar(){
    //con el keyword this llamar al metodo setEnfermo(0)
}

File: Java/POO_Java/src/com/codeismo/animales/Animal.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */

package com.codeismo.animales;

import java.util.Date;

/**
 *
 * @author Aula E3
 */
public abstract class Animal {
    /*Definir los atributos de la clase*/
    /*Definir los metdos de la clase*/
    private String nombre;
    private Date fechaNacimiento;
    private boolean enfermo;//true o false
    /*Constructor*/
    public Animal(){
        this.nombre = "Fido";
        this.enfermo = false;
        this.fechaNacimiento = new Date();
    }

    public Animal(String nombre, Date fechaNacimiento, boolean enfermo) {
        this.nombre = nombre;
        this.fechaNacimiento = fechaNacimiento;
        this.enfermo = enfermo;
    }
    
    
    /*Declaracion de Métodos*/
    public void sanar(){
        this.enfermo = false;
    }
    public void enfermar(){
        this.enfermo = true;
    }
    public void imprimirDatos(){
        System.out.println("Nombre: " + this.nombre);
        System.out.println("Fecha de nacimiento" + this.fechaNacimiento.toString() );
    }

    /** Getters y Setters **/
    public String getNombre() {
        return nombre;
    }

    public Date getFechaNacimiento() {
        return fechaNacimiento;
    }

    public boolean isEnfermo() {
        return enfermo;
    }

    public void setNombre(String nombre) {
        this.nombre = nombre;
    }

    public void setFechaNacimiento(Date fechaNacimiento) {
        this.fechaNacimiento = fechaNacimiento;
    }

    public void setEnfermo(boolean enfermo) {
        this.enfermo = enfermo;
    }  
    /*Metodo Abstracto
    No se implementa*/
    public abstract void hablar();
}
